.github/workflows/ci.yml

```
@@ -325,7 +325,7 @@ jobs:
           python builder.pyz build -p ${{ env.PACKAGE_NAME }} --cmake-extra=-DBUILD_SHARED_LIBS=ON
 
   osx:
-    runs-on: macos-latest
+    runs-on: macos-13
```

As Github has more OS version support, we can run on multiple OS versions using the matrix, like what we did for linux. 

```suggestion
    runs-on: ${{ matrix.runner }}
    strategy:
      matrix:
        runner:
          - macos-13
          - macos-latest
```

---

`macos-latest` currently fails. In the future, yes, that's a good idea.

---

I saw a workaround there https://github.com/orgs/Homebrew/discussions/5159
It is kind of hack, while I think it should be fine as we just use it in CI tests.